### PR TITLE
Expose PluralOperands::from_str() over capi FFI

### LINKED ISSUE #794: Expose PluralOperands::from_str() over capi FFI
It would be convenient to be able to create an `ICU4XPluralOperands` from a string representation of a floating-point number. We should expose the `PluralOperands::from_str()` functionality over FFI.

### PR DESCRIPTION
**Pull request by nordzilla**
Fixes #794 

- Add new struct for ICU4XCreatePluralOperandsResult
- Add new function for icu4x_plural_operands_create()

### CODE REVIEW

File: ffi/capi/src/pluralrules.rs
Comment (Manishearth) [inline ffi/capi/src/pluralrules.rs:83]: issue: we should probably handle this before landing

**Review by Manishearth — DISMISSED** [review]
r=me with the todo fixed

File: ffi/capi/src/pluralrules.rs
Comment (Manishearth) [inline ffi/capi/src/pluralrules.rs:172]: IMO we should convert PluralOperands to ICU4XPluralOperands and not use a From impl for this final conversion to the error type so that the failure case can be handled

File: ffi/capi/src/pluralrules.rs
Comment (nordzilla) [reply ffi/capi/src/pluralrules.rs]: Ah sorry, I think that I had a misunderstanding here. 

I looked at the `icu4x_create_locale` as an example, which has this same `todo` item. 
https://github.com/unicode-org/icu4x/blob/a0d4cf41951316c7f1c8733e20e8577fcd7e3f51/ffi/capi/src/locale.rs#L24

I took that to mean "return real error types instead of a boolean", but I see now that `icu4x_create_locale` just returns a `Locale` object regardless, and not a `Result` type with a `success` boolean. 

I believe that I am handling the errors to the extent that we currently handle them in this code, which is a boolean flag. 

I will remove the comment, since I believe it is not relevant, but I believe that this code should be sufficient unless I am misunderstanding you.

Comment (nordzilla) [reply ffi/capi/src/pluralrules.rs]: Actually, let me just do that anyway. I prefer to be explicit about things.

**Review by Manishearth — APPROVED** [review]

**Comment by Manishearth** [comment]
LGTM! Would be nice if the C++ side also existed, but no rush on that.

**Comment by nordzilla** [comment]
> LGTM! Would be nice if the C++ side also existed, but no rush on that.

I will file a follow-up issue for these, but I'm a bit pressed for time at the moment, and only need the `capi` version.

**Pull request merged**
